### PR TITLE
Feat: Remove failed attachments uploads

### PR DESCRIPTION
**Pull request by khushal87**
# Submit a pull request

## CLA

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [x] The code changes follow best practices
- [x] Code changes are tested (add some information if not applicable)

## Description of the pull request

Fix the issue "Cannot remove failed to upload attachments".

Reference - https://stream-io.atlassian.net/browse/CRNS-393

### CODE REVIEW

File: package/src/components/MessageInput/FileUploadPreview.tsx [hunk 1]
```diff
@@ -113,72 +117,74 @@ const FileUploadPreviewWithContext = <
...
-                marginBottom: 0,
-              }
-            : {},
-          {
-            borderColor: grey_whisper,
-            width: flatListWidth - 16,
-          },
-          fileContainer,
-        ]}
+    <View>
```
Comment (vishalnarkhede) [inline package/src/components/MessageInput/FileUploadPreview.tsx:120]: This can be replaced with `<>`

**Review by vishalnarkhede — APPROVED** [review]

File: package/src/components/MessageInput/FileUploadPreview.tsx [hunk 2]
```diff
@@ -113,72 +117,74 @@ const FileUploadPreviewWithContext = <
...
-        ]}
+    <>
+      <UploadProgressIndicator
+        action={() => {
+          uploadFile({ newFile: item });
+        }}
+        active={item.state !== FileState.UPLOADED && item.state !== FileState.FINISHED}
+        style={styles.overlay}
+        type={
+          item.state === FileState.UPLOADING
```
Comment (madsroskar) [inline package/src/components/MessageInput/FileUploadPreview.tsx:128]: Just a personal preference thing, but might it be a bit easier to read this if the nested ternary were to be extracted into a variable or something? Or maybe even if it's just wrapped in parentheses?

Comment (madsroskar) [reply package/src/components/MessageInput/FileUploadPreview.tsx:128]: I realize this wasn't introduced in this PR - don't mind me for the sake of getting this merged :)

File: package/src/components/MessageInput/FileUploadPreview.tsx [hunk 3]
```diff
@@ -113,72 +117,74 @@ const FileUploadPreviewWithContext = <
...
+            ? ProgressIndicatorTypes.IN_PROGRESS
+            : item.state === FileState.UPLOAD_FAILED
+            ? ProgressIndicatorTypes.RETRY
+            : undefined
+        }
```
Comment (madsroskar) [inline package/src/components/MessageInput/FileUploadPreview.tsx:133]: I'm not sure about what naming makes proper sense here, but I mean something along these lines. `shouldAllowRetry` should probably have a different name if you can think of something more descriptive, I think.
```suggestion
        type={() => {
          const shouldAllowRetry = item.state === FileState.UPLOAD_FAILED
            ? ProgressIndicatorTypes.RETRY
            : undefined;
          
          return item.state === FileState.UPLOADING
            ? ProgressIndicatorTypes.IN_PROGRESS
            : shouldAllowRetry;
        }}
```

**Comment by vishalnarkhede** [comment]
One more thing, I think the commit messages don't use appropriate [conventional commit prefix](https://www.conventionalcommits.org/en/v1.0.0-beta.4/#summary). e.g.,

- `Feat: Remove failed attchments uploads`: This needs to be a `fix: ...` since its not a new feature, we are just fixing that UX
- `fix: code refactor and fix tests`: This needs to be a `refactor: ...`
- `fix: tests for message.js `: This needs to be a `test: ...`

I will recommend closing this PR and opening a separate PR, with appropriate commit message(s). It's important to maintain a good commit history because our release system is dependent on how well we manage commits.

**Pull request closed without merging**